dev/io.openliberty.microprofile.telemetry.1.0.internal/src/io/openliberty/microprofile/telemetry/internal/cdi/OpenTelemetryProducer.java

```diff
@@ -131,20 +131,16 @@ private boolean checkDisabled(Map<String, String> oTelConfigs) {
     private HashMap<String, String> getTelemetryProperties() {
         HashMap<String, String> telemetryProperties = new HashMap<>();
         for (String propertyName : config.getPropertyNames()) {
-            if (propertyName.startsWith("otel.") || propertyName.startsWith("OTEL_")) {
-                config.getOptionalValue(propertyName, String.class).ifPresent(
-                                                                              value -> telemetryProperties.put(propertyName, value));
-
+            if (propertyName.startsWith("otel.")) {
+                telemetryProperties.put(propertyName,config.getValue(propertyName, String.class));
             }
```

Comment:
I suspect this fails for an environment variable named `otel_service_name` if `otel.service.name` is not also defined in another config source.

In most scenarios you do get away with it:
- if `otel.service.name` is defined in any config source, then we'll look its value up from MP Config and we'll find `otel_service_name` in the environment.
- if `OTEL_SERVICE_NAME` is defined in the environment and nowhere else then we won't add it here, but OTel's own config logic will find it.

I think the nearest thing we can get to correct is:
> If (propertyName starts with `otel` ignoring case) {
>   look up the value and add it to the map
>   lowercase the name and convert `_` to `.`
>   look up that name, if it has a value then add that to the map
> }

This assumes that all properties start `otel` and contain only lowercase alphanumeric characters and `.` (which I _think_ is good enough, I think some properties contain `-` but OTel will accept `.` instead)

Reply:
I think you also have a problem here if a config property is set in one config source, but then unset by a config source with a higher priority.

E.g. `foo=bar` in the environment, and then `foo=` as a system property.

I think in this case, `config.getPropertyNames()` will include `foo`, but `config.getValue("foo")` will throw an exception.